File: core/src/main/java/meshu/core/l1/FrameKind.java
package meshu.core.l1;

/**
 * L1 frame kinds (PROTOCOL.md §3.1). The on-wire first byte is
 * {@code VVKKKKKK} — version in the top 2 bits (0b00 for v1), kind in the low 6.
 */
public final class FrameKind {

    public static final int DATA = 0x10;   // L2 payload fragment
    public static final int ACKBM = 0x12;  // acknowledge with received-frame bitmap
    public static final int NACK = 0x13;   // reject message
    public static final int ABORT = 0x14;  // abandon an in-progress message
    public static final int PING = 0x15;   // liveness / path probe

    private FrameKind() {
    }

    /** NACK / ABORT reason codes (§3.8). */
    public static final class Reason {
        public static final int UNSUPPORTED_VERSION = 0x01;
        public static final int MSG_TOO_LARGE = 0x02;
        public static final int DECRYPT_FAILED = 0x03;
        public static final int REASM_TIMEOUT = 0x04;
        public static final int BUSY = 0x05;
        public static final int MALFORMED = 0x06;

        private Reason() {
        }
    }
}

File: wallet/src/l1/frameKind.ts
/**
 * L1 frame kinds (PROTOCOL.md §3.1). On-wire first byte is VVKKKKKK:
 * version (top 2 bits, 0b00 for v1) ‖ kind (low 6 bits). TS port of FrameKind.
 */
export const KIND_DATA = 0x10;
export const KIND_ACKBM = 0x12;
export const KIND_NACK = 0x13;
export const KIND_ABORT = 0x14;
export const KIND_PING = 0x15;

/** NACK / ABORT reason codes (§3.8). */
export const REASON_UNSUPPORTED_VERSION = 0x01;
export const REASON_MSG_TOO_LARGE = 0x02;
export const REASON_DECRYPT_FAILED = 0x03;
export const REASON_REASM_TIMEOUT = 0x04;
export const REASON_BUSY = 0x05;
export const REASON_MALFORMED = 0x06;
